tiledb/sm/array_schema/dimension.cc

```diff
@@ -302,9 +305,31 @@ void Dimension::crop_range(Range* range) const {
   crop_range_func_(this, range);
 }
 
+template <class T>
+uint64_t Dimension::domain_range(const Range& range) {
+  assert(!range.empty());
+
+  if (&typeid(T) == &typeid(float) || &typeid(T) == &typeid(double))
+    return 0;
+
+  auto r = (const T*)range.data();
+  uint64_t ret = r[1] - r[0];
+  if (ret == std::numeric_limits<uint64_t>::max())  // overflow
```

Comment:
Although you are right, what we want to ensure here is that `ret` (of type `uint64_t`) does not overflow when we do `r[1] - r[0] + 1`. To capture the overflow, we first do `r[1] - r[0]` which we know can't overflow. Then, the only way this can overflow is if its value is equal to `MAX_UNT64` before we add the final `1`.

Reply:
Ok, I think I understand, but want to verify: this is not checking whether the domain dtype will overflow its range, but whether essentially the cell count (as `uint64_t`) will overflow?

Reply:
The latter, correct.